includes/class-sensei-utils.php

```diff
@@ -292,43 +292,21 @@ public static function sensei_delete_activities( $args = array() ) {
 	} // End sensei_delete_activities()
 
 	/**
-	 * Delete all activity for specified user
+	 * Delete all activity for specified user.
 	 *
 	 * @access public
 	 * @since  1.5.0
-	 * @param  integer $user_id User ID
+	 *
+	 * @deprecated 3.0.0 Use `\Sensei_Learner::delete_all_user_activity` instead. But if you want also remove the enrolment terms, use `\Sensei_Learner::delete_user_registers`.
+	 *
+	 * @param  integer $user_id User ID.
 	 * @return boolean
 	 */
 	public static function delete_all_user_activity( $user_id = 0 ) {
+		_deprecated_function( __METHOD__, '3.0.0', 'To remove only activities use `\Sensei_Learner::delete_all_user_activity`. To remove also enrolment terms, use `\Sensei_Learner::delete_user_registers`' );
```

Comment:
Note that the 3rd parameter to this function should be the function name; otherwise, you'll see the following in the logs:

```
PHP Notice:  Sensei_Utils::delete_all_user_activity is <strong>deprecated</strong> since version 3.0.0! Use To remove only activities use `\Sensei_Learner::delete_all_user_activity`. To remove also enrolment terms, use `\Sensei_Learner::delete_user_registers` instead.
```

Reply:
Thank you for this catch! TIL =)

As suggested, I refactored to use all the logic inner the `delete_all_user_activity`. So I updated the deprecations replacement to only this method, fixing it ;)
 https://github.com/Automattic/sensei/pull/2898/commits/b969b0af5dd94fcaa1d1924f2761b62542e14a6d